sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -644,3 +647,46 @@ async fn test_encoding_edge_cases(ctx: TestContext) -> Result<(), Box<dyn Error>
 
     Ok(())
 }
+
+#[recorded::test]
+async fn test_immutability_policy(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    //TODO: Point to versioning account, add datetimes to recording variables
+    // Recording Setup
+    let recording = ctx.recording();
+    let container_client = get_container_client(recording, false).await?;
+    let blob_client = container_client.blob_client(&get_blob_name(recording));
+    container_client.create_container(None).await?;
+    create_test_blob(&blob_client, None, None).await?;
+    let test_expiry_time = Some(OffsetDateTime::now_utc() + Duration::from_secs(300));
+    let test_immutability_policy_mode = ImmutabilityPolicyMode::Unlocked;
+
+    // Set Immutability Policy
+    let immutability_policy_options = BlobClientSetImmutabilityPolicyOptions {
+        immutability_policy_expiry: test_expiry_time.clone(),
+        immutability_policy_mode: Some(test_immutability_policy_mode),
+        ..Default::default()
+    };
+    let response = blob_client
+        .set_immutability_policy(Some(immutability_policy_options))
+        .await?;
+
+    // Assert
+    let mode = response.immutability_policy_mode()?;
+    let expires_on = response.immutability_policy_expires_on()?;
+
+    let response_mode: BlobImmutabilityPolicyMode = test_immutability_policy_mode.into();
+    assert_eq!(response_mode, mode.unwrap());
```

Comment:
Food for thought:
On the **request** side, we have `ImmutabilityPolicyMode`
On the **response** side, we have `BlobImmutabilityPolicyMode`

ImmutabilityPolicyMode is a sub-set of BlobImmutabilityPolicyMode, as the additional enum `Mutable` can be returned by the service.

So this raises the question:

- Leave implementation as-is, cx must track this subtlety between response/request model (I provided a `From` impl. so they can just call `.into()`)
- Flatten it so that on both request and response side we use `BlobImmutabilityPolicyMode`, fully knowing that we are going to offer an enum on the request side that will never work (`Mutable`)

Reply:
As discussed offline, combine to `ImmutabilityPolicyMode` which is the superset and document that `Mutable` is only intended for response side.